app/assets/javascripts/tests/_update_crowd_map_layer.test.js

```diff
@@ -86,6 +86,44 @@ test('when buildQueryParamsForAverages return false it does not request the aver
   t.end();
 });
 
+test('when the request for the averages succeeds and the app is on the mobile tab it passes a callback to drawRectangles', t => {
+  const map = {
+    clearRectangles: () => {},
+    drawRectangles: (_a, _b, callback) => { callback() }
+  };
+  const $http = mockHttp({ shouldFail: false });
+  const $location = { path: () => "/map_sessions" };
+  const rectangles ={
+    position: () => {}
+  };
+  const infoWindow = mock('show');
+  const service = _updateCrowdMapLayer({ map, $http, $location, infoWindow, rectangles });
+
+  service.call();
+
+  t.true(infoWindow.wasCalled());
+
+  t.end();
+});
+
+test('when the request for the averages succeeds and the app is on the mobile tab it passes a callback to drawRectangles with query params for averages', t => {
+  const map = {
+    clearRectangles: () => {},
+    drawRectangles: (_a, _b, callback) => { callback() }
+  };
+  const $http = mockHttp({ shouldFail: false });
+  const $location = { path: () => "/map_sessions" };
+  const infoWindow = mock('show');
+  const queryParamsForAverages = {};
+  const service = _updateCrowdMapLayer({ map, $http, $location, infoWindow });
+
+  service.call();
+
+  t.true(infoWindow.wasCalledWith2(queryParamsForAverages));
```

Comment:
There's a problem with this test. That I don't see the relationship between `buildQueryParamsForAverages.call` and `infoWindow.wasCalledWith2({})`. But you are almost there. You could also test `wasCalledWith3`